migtests/scripts/run-config-test.sh

```diff
@@ -0,0 +1,221 @@
+
+	cat ${EXPORT_DIR}/data/export_status.json || echo "No export_status.json found."
+	cat ${EXPORT_DIR}/metainfo/dataFileDescriptor.json
+
+	step "Verify the pg_dump version being used"
+	if [ "${SOURCE_DB_TYPE}" = "postgresql" ] && { [ -z "${BETA_FAST_DATA_EXPORT}" ] || [ "${BETA_FAST_DATA_EXPORT}" = "0" ]; }; then
+	    if ! grep "Dumped by pg_dump version:" "${EXPORT_DIR}/logs/yb-voyager-export-data.log"; then
+	        echo "Error: pg_dump version not found in the log file." >&2
+	    fi
+	fi
+
+	step "Fix data."
+	if [ -x "${TEST_DIR}/fix-data" ]
+	then
+		"${TEST_DIR}/fix-data" ${EXPORT_DIR}
+	fi
+
+	step "Create target database."
+	run_ysql yugabyte "DROP DATABASE IF EXISTS ${TARGET_DB_NAME};"
+	if [ "${SOURCE_DB_TYPE}" = "postgresql" ] || [ "${SOURCE_DB_TYPE}" = "oracle" ]; then
+		run_ysql yugabyte "CREATE DATABASE ${TARGET_DB_NAME} with COLOCATION=TRUE"
+	else
+		run_ysql yugabyte "CREATE DATABASE ${TARGET_DB_NAME}"
+	fi
+
+	step "Import schema."
+	yb-voyager import schema -c "${GENERATED_CONFIG}" --yes
+	run_ysql ${TARGET_DB_NAME} "\dt"
+
+	step "Run Schema validations."
+	if [ -x "${TEST_DIR}/validate-schema" ]
+	then
+		 "${TEST_DIR}/validate-schema"
+	fi
+
+	step "Import data."
+	yb-voyager import data -c "${GENERATED_CONFIG}" --yes
+	
+	step "Import remaining schema (FK, index, and trigger) and Refreshing MViews if present."
+	yb-voyager finalize-schema-post-data-import -c "${GENERATED_CONFIG}" --yes
+	
+	run_ysql ${TARGET_DB_NAME} "\di"
+	run_ysql ${TARGET_DB_NAME} "\dft" 
+
+	step "Run validations."
+	if [[ "${EXPORT_TABLE_LIST}" != "" && -x "${TEST_DIR}/validate-with-table-list" ]]; then
+		"${TEST_DIR}/validate-with-table-list"
+	elif [[ -x "${TEST_DIR}/validate" ]]; then
+		"${TEST_DIR}/validate"
+	fi
+
+	step "Run export-data-status"
+	yb-voyager export data status -c "${GENERATED_CONFIG}" --yes --output-format json
+
+	expected_file="${TEST_DIR}/export_data_status-report.json"
+	actual_file="${EXPORT_DIR}/reports/export-data-status-report.json"
+
+	if [ "${EXPORT_TABLE_LIST}" != "" ]
+	then
+		expected_file="${TEST_DIR}/export-data-status-with-table-list-report.json"
+	fi
+
+	step "Verify export-data-status report"
+	verify_report ${expected_file} ${actual_file}
+
+	step "Run import-data-status"
+	yb-voyager import data status -c "${GENERATED_CONFIG}" --yes --output-format json
+
+	expected_file="${TEST_DIR}/import_data_status-report.json"
+	actual_file="${EXPORT_DIR}/reports/import-data-status-report.json"
+
+	if [ "${EXPORT_TABLE_LIST}" != "" ]
+	then
+		expected_file="${TEST_DIR}/import-data-status-with-table-list-report.json"
+	fi
+
+	verify_report ${expected_file} ${actual_file}
+
+	step "End Migration: clearing metainfo about state of migration from everywhere."
+	yb-voyager end migration -c "${GENERATED_CONFIG}" --yes
+
+	step "Clean up"
+	./cleanup-db
+	rm -rf "${EXPORT_DIR}"
+	run_ysql yugabyte "DROP DATABASE IF EXISTS ${TARGET_DB_NAME};"
+	rm -rf "${GENERATED_CONFIG}"
+}
+
+main
```

Comment:
I guess the entire script must be essentially the same as the normal run-test.sh script except the config generation part and the commands themselves?
Same for the other scripts?

Reply:
Yes correct